packages/grafana-data/src/types/explore.ts

```diff
@@ -18,3 +20,16 @@ export interface ExploreUrlState<T extends DataQuery = AnyQuery> {
 export type SplitOpen = <T extends DataQuery = any>(
   options?: { datasourceUid: string; query: T; range?: TimeRange } | undefined
 ) => void;
+
+/**
+ * Object with this interface should be returned by getLogsVolumeDataProvider
+ * if such method is implemented in an DataSourceApi object.
+ *
+ * It's used internally in Explore only at the moment but could be moved to
+ * @grafana/data and DataSourceApi if it's implemented in other data sources.
+ *
+ * @internal
+ */
+export interface RelatedDataProvider {
+  getData(): Observable<DataQueryResponse>;
+}
```

Comment:
i might be wrong here, but as far as I understood from the initial discussions, the apis were going to be something like 

```
type Providers = 'logsVolume' | 'somethingElse';
interface DataSourceApi  ...{
  // other stuff...
  relatedDataProviders?: {
    [key in Providers]?: (DataQueryRequest<TQuery>) => Observable<DataQueryResponse>;
  }
}
```

in DataSourceApi, so that we were not going to create Explore specific APIs.

am I missing something?

Reply:
After discussing it with @aocenas we decided that it'd be premature to decide about more generic API now without having specific use cases for `... | 'somethingElse';`.  We thought this API can change in numerous ways depending on specific requirements. The way it's implemented at the moment can easily scale to what you mention above without making any decisions upfront.

It's not intended to have specific methods in DataSourceApi, but make it more generic once we have a use case for it.

Reply:
Added a note on this in b77d2f105a7a8bf3748fea8021daadaf2ba69556

Reply:
so we don't want to expose this to the world yet? makes totally sense, this will give us freedom to break stuff while we figure it out 👍